### ISSUE TITLE
Arduino RC Switch - Format for RC codes

### ISSUE THREAD

**Issue by dooney100** [ni-c/heimcontrol.js#129]
I have sniffed remote controls and they are 333107 etc i.e. only 6 digits. When they are converted into binary with python2 then they don't seem to match the requested format in heimcontrol.js interface.
Any ideas - do I need to add extra digits to the remote codes to make them 8 digits?

**Comment by tjanson** [comment]
How did you sniff the codes? Did you use the [RC-Switch `ReceiveDemo` script](https://code.google.com/p/rc-switch/wiki/HowTo_Receive)?
Also, what do the switches on your outlet look like (compare with [this](https://code.google.com/p/rc-switch/wiki/HowTo_OperateLowCostOutlets))?

**Comment by dooney100** [comment]
I used the ReceiveDemo_simple.pde file on arduino. I am using this remote for power sockets - http://www.amazon.co.uk/gp/product/B00JTRSG88?psc=1&redirect=true&ref_=oh_aui_detailpage_o08_s00

**Comment by tjanson** [comment]
Unfortunately it’s quite possible that this remote’s code protocol is unsupported. It might help to see the exact output of the advanced ReceiveDemo script, but I’m afraid you are the first with this kind of remote.

Have you been able to replicate the codes with the RC-Switch library (on its own, without Heimcontrol) (or any other software, for that matter)?

**Comment by dooney100** [comment]
thanks for your response - I'll paste the output of ReceiveDemo script in a separate message. Below is the message from heimcontrol when i try to enter the remote codes. I thought it might be something to do with heimcontrol software just requesting a minimum number of digits for entry into the form which could be changed somewhere in the heimcontrol code? If I add extra digits to the form entry then it accepts the code ok.

![image](https://cloud.githubusercontent.com/assets/16168747/11636902/64713dc0-9d16-11e5-9c14-2da8d3d52a4a.png)

**Comment by tjanson** [comment]
I would try to make sure sending the code works _somehow_, e.g., using
[`SendDemo`](https://github.com/sui77/rc-switch/blob/master/examples/SendDemo/SendDemo.pde#L44-L48)
(in binary mode, no protocol abstraction) or similar.
That would verify that you have the code itself as well as the pulse length
correctly, that the transmitter actually works, and so on.

Once that’s done, getting it working within Heimcontrol will be easy.

**Comment by dooney100** [comment]
ok - seems like rc controls are 24 bit hardcoded in heimcontrol. I think this can be changed in /plugins/arduino/views/settings.jade

**Comment by tjanson** [comment]
No, they are not hardcoded. Again, I suggest you take things step by step and get RC-Switch working by itself, _then_ tackle Heimcontrol integration. I’ll be glad to help if you have specific questions.

(As a side note, going via Arduino isn’t necessary – the Raspberry Pi’s own GPIOs can drive the 433 MHz transmitter too.)
